fix: exit with 0 if re-publishing the same library (#2106)
canonical/charmcraft: docs/reference/changelog.rst

  For a complete list of commits, see the `X.Y.Z`_ release on GitHub.

3.3.1 (2025-01-23)
------------------

This release contains some bug fixes for the 3.3 branch.

Command line
============

> - the :ref:`ref_commands_publish-lib` command now returns ``0`` if the libraries
>   published match the versions already on the store. It also does not create a new
>   revision on the store in this case.

Reviewer: ```suggestion
- The :ref:`ref_commands_publish-lib` command now returns ``0`` if the libraries
  published match the versions already on the store. It also no longer creates a new
  revision on the store in this case.
```

Documentation
=============

> - Adds :doc:`reference </reference/platforms>` and a
>   :ref:`how-to guide <select-platforms>` for the ``platforms`` keyword in the
>   :ref:`charmcraft-yaml-file`.
> - Fixes for some links to the `Juju`_ documentation.

Reviewer: ```suggestion
- Add :doc:`reference </reference/platforms>` and a
  :ref:`how-to guide <select-platforms>` for the ``platforms`` key in the
  :ref:`charmcraft-yaml-file`.
- Fix some links to the `Juju`_ documentation.
```
